mat64/dense_arithmetic.go

```diff
@@ -30,17 +30,17 @@ func (m *Dense) Add(a, b Matrix) {
 	bMat, _ := untranspose(b)
 	m.reuseAs(ar, ac)
 	var restore func()
-	if m == aMat {
-		m, restore = m.isolatedWorkspace(aMat)
-		defer restore()
-	} else if m == bMat {
-		m, restore = m.isolatedWorkspace(bMat)
-		defer restore()
-	}
 
-	if a, ok := a.(RawMatrixer); ok {
-		if b, ok := b.(RawMatrixer); ok {
-			amat, bmat := a.RawMatrix(), b.RawMatrix()
+	if arm, ok := a.(RawMatrixer); ok {
+		if brm, ok := b.(RawMatrixer); ok {
+			amat, bmat := arm.RawMatrix(), brm.RawMatrix()
```

Comment:
Given the algorithm, we don't need to isolate the workspace, just check overlap, correct?